tests/e2e/tests/CommonTestMethods.ts

```diff
@@ -0,0 +1,99 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { NewWorkspace, CLASSES, Ide, ProjectTree, TestConstants, QuickOpenContainer, Terminal, TopMenu, Dashboard, Editor } from '..';
+import { e2eContainer } from '../inversify.config';
+import { error } from 'selenium-webdriver';
+
+const ide: Ide = e2eContainer.get(CLASSES.Ide);
+const projectTree: ProjectTree = e2eContainer.get(CLASSES.ProjectTree);
+const namespace: string = TestConstants.TS_SELENIUM_USERNAME;
+const quickOpenContainer: QuickOpenContainer = e2eContainer.get(CLASSES.QuickOpenContainer);
+const terminal: Terminal = e2eContainer.get(CLASSES.Terminal);
+const topMenu: TopMenu = e2eContainer.get(CLASSES.TopMenu);
+const dashboard: Dashboard = e2eContainer.get(CLASSES.Dashboard);
+const editor: Editor = e2eContainer.get(CLASSES.Editor);
+const newWorkspace: NewWorkspace = e2eContainer.get(CLASSES.NewWorkspace);
+
+export function createAndOpenWorkspace(workspaceName : string, stack: string) {
+    suite('Create ' + stack + ' workspace ' + workspaceName, async () => {
```

Comment:
-1 For that. First of all as far as I remember this is described as a bed approach in the mocha documentation. It may lead to errors.
Another point, I strongly disagree to mix everything in one place.
I follow an approach that one class should be responsible for one thin.
For example we already have "runTask" method in the "TopMenu" pageobject.

Reply:
This class is responsible for common test methods, something like common tests used in stack tests. So it makes sense to me to have all the "test methods" in one place. If there is a `runTask` in TopMenu, than we can remove it from here and use the one from TopMenu of course, I'll fix that in the next commit. Thank you for the hint. :+1:

Reply:
There is a big difference between common methods grouped by tests type (stacks tests), and common methods that aggregate different pageobjects and methods without grouping.
On my opinion, at least, they should be separated to the 2 different classes (common ide, and common dashboard):
 - createAndOpenWorkspace (dashboard)
 - stopAndRemoveWorkspace (dshboard)
 - waitIdeAndProjectImported (ide)
On my opinion, at least, they should be separated to the 2 different classes (common ide, and common dashboard)
And next methods looks as overengineering for me:
 - runTaskAndCloseTerminal
 - openFileInAssociatedWorkspace
 - runTask

Reply:
Can you please send me docs you are talking about? I found just this https://mochajs.org/#dynamically-generating-tests and according to it, it seems that using parameters in the description is OK. Thank you

Reply:
Ok, agree about parameters.
Let's discuss the second topic.